### ISSUE TITLE
Documentation for WeightedMSECriterion

### ISSUE THREAD

**Issue by jonathantompson** [torch/nn#384]
This is less of an issue, but more of a request.  What is WeightedMSECriterion actually trying to compute?

I would have thought it would be:

crit = sum_{i=1}^n (w_i * (input_i - target_i)^2)

But it seems to be:

crit = sum_{i=1}^n ((input_i - w_i * target_i)^2)

What is this?  Am I wrong?  Am I going crazy?  Should I fix this so that is has linear weighting (as one might expect)?

**Comment by joeyhng** [comment]
I've also tried to use it recently and I was thinking the same thing as you. I don't know if this is intended or a bug.

**Comment by jonathantompson** [comment]
Yeah, I have a hard time believing this was intentional... Although maybe koray can help clarify.

**Comment by soumith** [comment]
cc: @koraykv what would you like done in this case?

**Comment by koraykv** [comment]
I think this is doing what it is supposed to be doing, I remember using
this for deconvolutional nets where the border effects to counter the
border effects. The weights are acting like a mask on the targets. We
should add doc clarifying that weights are on the targets, not a very good
choice of naming I guess.

koray

**Comment by canghel** [comment]
Sorry, is there a version of MSECriterion that weighs the error as described by jonathantompson?

**Comment by jonathantompson** [comment]
Here you go:

https://gist.github.com/jonathantompson/e4701e28286a04606c445994e3b2debd

and here is the unit test:

https://gist.github.com/jonathantompson/f7956373c7fd509020c38721c88a8044

**Comment by canghel** [comment]
Wow, thank you!  You made my day!

**Comment by jonathantompson** [comment]
NP :-)
